File: algorithm/src/main/java/baekjoon/p02947/Main.java
package baekjoon.p02947;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int[] inputs = new int[5];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextInt();
        }

        for (int[] solve : solve(inputs)) {
            for (int n : solve) {
                System.out.print(n + " ");
            }
            System.out.println();
        }
    }

    private static List<int[]> solve(int[] inputs) {
        List<int[]> ret = new ArrayList<>();
        int[] arr = inputs;
        int len = inputs.length;

        for (int i = 0; i < len - 1; i++) {
            for (int j = 0; j < len - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    arr = arr.clone();
                    swap(arr, j, j + 1);
                    ret.add(arr);
                }
            }
        }

        return ret;
    }

    private static void swap(int[] arr, int a, int b) {
        int temp = arr[a];
        arr[a] = arr[b];
        arr[b] = temp;
    }

}

File: algorithm/src/main/java/baekjoon/p02947/Main.kt
package baekjoon.p02947

fun main(args: Array<String>) {
    val inputs = readLine()!!.split(" ").map { it.toInt() }
    for (solve in solve(inputs)) {
        println(solve.joinToString(" "))
    }
}

private fun solve(inputs: List<Int>): List<IntArray> {
    val ret = ArrayList<IntArray>()
    var arr = inputs.toIntArray()

    for (i in 0 until inputs.size - 1) {
        for (j in 0 until inputs.size - 1 - i) {
            if (arr[j] > arr[j + 1]) {
                arr = arr.clone()
                swap(arr, j, j + 1)
                ret.add(arr)
            }
        }
    }

    return ret
}

private fun swap(arr: IntArray, a: Int, b: Int) {
    val temp = arr[a]
    arr[a] = arr[b]
    arr[b] = temp
}